src/lib/php/Dao/ClearingDao.php

```diff
@@ -1022,4 +1022,24 @@ public function getTotalDecisionCount($uploadId, $groupId)
     $foundCounter = $this->dbManager->getSingleRow($sql, $params, $statement);
     return $foundCounter['cnt'];
   }
+
+  /**
+   * @param uploadTreeId
+   * @return count
+   */
+  public function getCandidateCountForItemId($uploadTreeId)
```

Comment:
Can you please add this check to `createDecisionFromEvents()` as well so we can make sure the check happens at backend as well?

https://github.com/fossology/fossology/blob/b5a389be8ef413fc488b56700ea23bdf3ff9c73a/src/lib/php/Dao/ClearingDao.php#L334

Reply:
Also for `markDirectoryAsDecisionType()`

https://github.com/fossology/fossology/blob/b5a389be8ef413fc488b56700ea23bdf3ff9c73a/src/lib/php/Dao/ClearingDao.php#L753